### PR TITLE
Feature login signup

### PR DESCRIPTION
**Pull request by yashkosambia14**
Guys please review, I have completed signup and login 
which is now functional, please check for any changes and I will keep updating the pages required for it.

### CODE REVIEW

File: data/loginUsers.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    throw { code: 400, error: `Username must be at least 4 characters long` };
+
+  if (typeof password != "string")
+    throw { code: 400, error: `Password needs to be a string` };
+  let myPassword = password.split(" ");
+  if (myPassword.length > 1)
+    throw { code: 400, error: `Password cannot have spaces` };
+  if (password.length < 6)
+    throw { code: 400, error: `Password must be at least 6 characters long` };
+}
```
Comment (daniprogrammi) [inline data/loginUsers.js]: This should be in your validation file which you're already importing, so you should just be calling this function from 'login_validations' wherever you're using it in this file

File: data/loginUsers.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+async function checkInput(username, password) {
+  if (!username || !password)
+    throw { code: 400, error: `Username or Password cannot be empty` };
+  if (typeof username != "string")
+    throw { code: 400, error: `Username needs to be a string` };
+  let myUsername = username.split(" ");
+  if (myUsername.length > 1)
+    throw { code: 400, error: `Username cannot have spaces` };
+  if (/[^\w\s]/.test(username))
```
Comment (daniprogrammi) [inline data/loginUsers.js]: I think this regex should just be /[^\w]/ You're testing if the username contains any character that's not \w - (a-z,A-Z,0-9 or _). A username shouldn't have spaces so \s shouldn't be there in the test regex

File: routes/loginRoutes.js [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+  if (!lastName || typeof lastName != "string")
+    throw { code: 400, error: "Names cam only be strings" };
+  let myFirstName = firstName.split(" ");
+  let myLastName = lastName.split(" ");
+  if (myFirstName.length > 1 || myLastName.length > 1)
+    throw { code: 400, error: `Name cannot have spaces` };
+  if (/[^\w\s]/.test(firstName && lastName))
+    throw { code: 400, error: `Name can only be alphanumeric characters` };
+}
+// ERROR CHECKING ENDS
```
Comment (daniprogrammi) [inline routes/loginRoutes.js]: Should be in a validation file

File: views/pages/private.handlebars [hunk 4]
```diff
@@ -0,0 +1,5 @@
+<div class="container">
+    <h1>Welcome{{username}}!</h1><br>
+    <img src="../iamges/LOGO.jpg" class="img-thumbnail" alt="...">
```
Comment (daniprogrammi) [inline views/pages/private.handlebars]: typo in src -- should be images

File: views/pages/signup.handlebars [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+        <label for="username">Username:</label>
+        <input type="text" name="username" id="username" placeholder="Enter your username"><br>
+        <label for="password">Password:</label>
+        <input type="password" name="password" id="password" placeholder="Enter your password"><br>
+        <label for="password">Confirm Password:</label>
+        <input type="password" name="password2" id="password2" placeholder="Enter your password"><br>
+        <label for="age">First Name:</label>
+        <input type="text" name="firstName" id="firstName" placeholder="Enter your First name"><br>
+        <label for="age">Last Name:</label>
+        <input type="text" name="lastName" id="lastName" placeholder="Enter Last name "><br>
```
Comment (daniprogrammi) [inline views/pages/signup.handlebars]: For the signup page, I think the only thing we will need will be a username, password and confirm password. Everything else should be entered in a profile edit page I think

**Review by daniprogrammi — COMMENTED** [review]
Can you move all of the files from views/pages to views/frames and update the routes accordingly?

File: routes/loginRoutes.js (hunk 3 above)
Comment (yashkosambia14) [reply routes/loginRoutes.js]: completed

File: data/loginUsers.js (hunk 1 above)
Comment (yashkosambia14) [reply data/loginUsers.js]: Yeah, I just kept it I was going to remove it anyways.

Comment (yashkosambia14) [reply data/loginUsers.js]: completed

File: views/pages/signup.handlebars (hunk 5 above)
Comment (yashkosambia14) [reply views/pages/signup.handlebars]: Yes, Initially it was made like that, but after discussion with Harish added more fields,

File: data/loginUsers.js (hunk 2 above)
Comment (yashkosambia14) [reply data/loginUsers.js]: got it

**Review by SaiHarishVitta — APPROVED** [review]

**Review by daniprogrammi — APPROVED** [review]

**Pull request merged**